src/controller/c/robot.c

```diff
@@ -1170,22 +1166,22 @@ static char *compute_socket_filename() {
       free(folder);
       return NULL;
     }
-    if (count > 1)  // sort folders (robot names) in alphabetical order
-      qsort(filenames, count, sizeof(char *), strcmp_sort);
-    for (int i = 0; i < count; i++) {  // keep only the first extern controller
+    if (count > 1) {  // more than one extern controller in the current instance of Webots
+      fprintf(stderr, "Webots instance %d has several extern controller robots.\n", number);
+      fprintf(stderr, "Please set the WEBOTS_CONTROLLER_URL environment variable to select one among:\n");
```

Comment:
I have spotted a bug when displaying the possible robots to which the user can connect. If we open multiple worlds with different robot names successively, folder are created but never deleted. Therefore, the list contains a lot of names which are not available in the current world. 

Also the current behavior doesn't check if the robot is waiting for an extern controller, therefore all robots are considered as "waiting for extern controller". It makes it impossible to work on worlds with multiple robots like `/projects/samples/devices/worlds/gps` for example.

Reply:
Well spotted! I believe I fixed this problem.